### PR TITLE
[v8.12] Document incompatibility with OCaml 4.12.

### PR DESCRIPTION
**Pull request by Zimmi48**
Following up of @silene's comment: https://github.com/coq/coq/pull/12894#issuecomment-681878128

**Kind:** documentation

### CODE REVIEW

File: INSTALL.md
Comment (SkySkimmer) [inline INSTALL.md:12]:
```suggestion
  All Coq versions up to and including Coq 8.12 should be considered incompatible
```
?

File: INSTALL.md
Comment (jfehrle) [inline INSTALL.md:10]: It's not been tested with 4.11?

File: INSTALL.md
Comment (jfehrle) [inline INSTALL.md:9]:
```suggestion
- [OCaml](https://ocaml.org/) (4.05.0 <= version < 4.12.0)
```

If you like.  Fine either way.

**Review by jfehrle — APPROVED** [review]
Looks good.  If you have no further changes, let me know and I'll merge it.

File: INSTALL.md
Comment (Zimmi48) [reply INSTALL.md:10]: 4.11.0 was just released. It doesn't look like we're testing it yet. @ejgallego @SkySkimmer might confirm. We should anyway "fix" this and make sure that Coq 8.12.1 is compatible with OCaml 4.11.0.

File: INSTALL.md
Comment (Zimmi48) [reply INSTALL.md:9]: It's more mathematical but less readable (?), so I don't know.

**Comment by SkySkimmer** [comment]
Should there be a check in configure too?

File: INSTALL.md
Comment (Zimmi48) [reply INSTALL.md:10]: I will wait for #12972 to land before rebasing this PR.

**Comment by Zimmi48** [comment]
Updated and now includes a configure check.

**Comment by jfehrle** [comment]
This is still a draft PR--is it ready for final review?  I'll let someone else review the config check.  (Anyone feel free to reassign to yourself.)

**Comment by Zimmi48** [comment]
Ignore this: it's just a test.

@coqbot: merge now

**Comment by Zimmi48** [comment]
This is still waiting for a review from @coq/build-maintainers / @coq/legacy-build-maintainers. I can't find his comment anymore, but I remembered @jfehrle saying other people should feel free to self-assign once the configure changes are reviewed.

**Comment by Zimmi48** [comment]
The change of target branch follows the discussion in #12894.

**Review by gares — APPROVED** [review]

**Pull request merged**
